src/crates-io/lib.rs

```diff
@@ -170,11 +171,12 @@ impl Registry {
         Ok(())
     }
 
-    pub fn search(&mut self, query: &str, limit: u8) -> Result<Vec<Crate>> {
+    pub fn search(&mut self, query: &str, limit: u8) -> Result<(Vec<Crate>, u32)> {
         let body = try!(self.req(format!("/crates?q={}&per_page={}", query, limit), None, Get,
                                  Auth::Unauthorized));
 
-        Ok(json::decode::<Crates>(&body).unwrap().crates)
+        let crates = json::decode::<Crates>(&body).unwrap();
```

Comment:
Oh holy cow I had no idea there were so many other `unwrap()` calls, guh yes that should be fixed but perhaps in the next PR.